### ISSUE TITLE
Spurious Failures in successful `require`

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#1332]
Need to run `t/fudgeandrun S11-modules/require.t` a few times to trigger this. The warnings reference [two successful `require`s](https://github.com/perl6/roast/blob/17e1dda1fb09a86b17375be9e9c022fbf09bb8b0/S11-modules/require.t#L41-L47), so there should not be any Failures.

This can be worked around with `(require ...) orelse Nil` or `(require ...) andthen Nil`. I also stuck a .^name print statement into `orelse` there and it always prints the module's name, not a Failure.

```
2017.12.11 zoffix@VirtualBox~/CPANPRC/rakudo (master)$ t/fudgeandrun S11-modules/require.t
t/spec/S11-modules/require.t .. 
1..34
ok 1 - successful require PACKAGE returns PACKAGE
ok 2 - successful require STRING returns STRING
ok 3 - require Test installs stub Test package at compile time
ok 4 - can introspect EXPORT of require'd package
ok 5 - can call our-sub from required module
ok 6 - can load InnerModule from a path at run time
ok 7 - can load InnerModule from a variable at run time
ok 8 - can import symbol not inside module
ok 9 - can load InnerModule by name and path, with import list
ok 10 - &bar didn't leak to outer scope
    1..2
    ok 1 - code dies
    ok 2 - right exception type (X::Import::MissingSymbols)
ok 11 - &-less import of sub does not produce `Null PMC access` error
WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
No such symbol 'InnerModule'
  in block <unit> at t/spec/S11-modules/require.t line 47

WARNING: unhandled Failure detected in DESTROY. If you meant to ignore it, you can mark it as handled by calling .Bool, .so, .not, or .defined methods. The Failure was:
No such symbol 'S11-modules::NoModule'
  in block <unit> at t/spec/S11-modules/require.t line 41

ok 12 - can call our-sub from required module
ok 13 - can load Fancy::Utilities at run time
ok 14 - can require with variable name
ok 15 - can call subroutines in a module by name
ok 16 - package returned from Fancy::Utilities matches the indirect lookup
ok 17 - require with import list
ok 18 - loading modules does not clobber GLOBAL
ok 19 - can change directory and require a module
ok 20 - requiring something non-existent does not make it segfault
    1..2
    ok 1 - code dies
    ok 2 - right exception type (X::Import::MissingSymbols)
ok 21 - throws correct exception
ok 22 - require works at BEGIN
ok 23 - require can import at BEGIN
ok 24 - got outer symbol
ok 25 - call method that causes a require
ok 26 - Did not find inner symbol
ok 27 - Cool::Utils has been merged
ok 28 - Cool::Utils was returned
ok 29 - 1. multiple requires with top level package already defined
ok 30 - 2. multiple requires with top level package already defined
ok 31 - 3. multiple requires with top level package already defined
ok 32 - class in required package
ok 33 - class in long required package name
ok 34 - require class with set constant
ok
All tests successful.
Files=1, Tests=34,  1 wallclock secs ( 0.03 usr  0.00 sys +  1.19 cusr  0.22 csys =  1.44 CPU)
Result: PASS
2017.12.11 zoffix@VirtualBox~/CPANPRC/rakudo (master)$ ./perl6 -v
This is Rakudo version 2017.12-15-g57d164b built on MoarVM version 2017.12-1-g912f967
implementing Perl 6.c.
2017.12.11 zoffix@VirtualBox~/CPANPRC/rakudo (master)$ 
```

Same symptoms in [RT#131996](https://rt.perl.org/Ticket/Display.html?id=131996#ticket-history). Closing it in favour of this Issue, since this one is easier to repro.

**Comment by zoffixznet** [comment]
Worth mentioning: this test file [on occasion flops](https://irclog.perlgeek.de/perl6-dev/search/?nick=&q=S11-modules%2Frequire.t) during stresstests

**Comment by niner** [comment]
Hopefully fixed in commit 666eb3abe7e61141073c31776cb0f43bac0a338c though the stated diagnosis was only really true for the first Failure. But with that fixed I fail to reproduce the others, too.
